src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp

```diff
@@ -71,46 +71,56 @@ ReturnCode_t TypeObjectRegistry::register_type_object(
+        EPROSIMA_LOG_ERROR(
+            XTYPES_TYPE_REPRESENTATION,
+            "Inconsistent CompleteTypeObject: " << exception.what());
         return eprosima::fastdds::dds::RETCODE_PRECONDITION_NOT_MET;
     }
 #endif // !defined(NDEBUG)
     TypeRegistryEntry complete_entry;
     TypeRegistryEntry minimal_entry;
     complete_entry.type_object.complete(complete_type_object);
     minimal_entry.type_object = build_minimal_from_complete_type_object(complete_type_object);
-    type_ids.type_identifier1(calculate_type_identifier(minimal_entry.type_object,
+    type_ids.type_identifier1(
+        calculate_type_identifier(
+            minimal_entry.type_object,
             minimal_entry.type_object_serialized_size));
-    type_ids.type_identifier2(calculate_type_identifier(complete_entry.type_object,
+    type_ids.type_identifier2(
+        calculate_type_identifier(
+            complete_entry.type_object,
             complete_entry.type_object_serialized_size));
     complete_entry.complementary_type_id = type_ids.type_identifier1();
     minimal_entry.complementary_type_id = type_ids.type_identifier2();
 
     std::lock_guard<std::mutex> data_guard(type_object_registry_mutex_);
     auto type_ids_result {local_type_identifiers_.insert({type_name, type_ids})};
-    auto min_entry_result {type_registry_entries_.insert({type_ids.type_identifier1(), minimal_entry})};
-    auto max_entry_result {type_registry_entries_.insert({type_ids.type_identifier2(), complete_entry})};
-    if (!type_ids_result.second || !max_entry_result.second)
+
+    if (type_ids_result.second)
     {
-        if (local_type_identifiers_[type_name] != type_ids ||
-                type_registry_entries_[type_ids.type_identifier1()] != minimal_entry ||
-                type_registry_entries_[type_ids.type_identifier2()] != complete_entry)
+        auto min_entry_result {type_registry_entries_.insert(
+                                   {type_ids.type_identifier1(), minimal_entry})};
+        if (!min_entry_result.second)
         {
-            if (type_ids_result.second)
-            {
-                local_type_identifiers_.erase(type_name);
-            }
-            if (min_entry_result.second)
-            {
-                type_registry_entries_.erase(type_ids.type_identifier1());
-            }
-            if (max_entry_result.second)
-            {
-                type_registry_entries_.erase(type_ids.type_identifier2());
-            }
-            return eprosima::fastdds::dds::RETCODE_BAD_PARAMETER;
+            EPROSIMA_LOG_INFO(
+                XTYPES_TYPE_REPRESENTATION,
+                "Type " << type_name << " already registered his EK_MINIMAL remotely.");
+        }
+        auto max_entry_result {type_registry_entries_.insert(
+                                   {type_ids.type_identifier2(), complete_entry})};
+        if (!max_entry_result.second)
+        {
+            EPROSIMA_LOG_INFO(
```

Comment:
Apart from warning the user, we could discuss that if the currently registered complete typeidentifier has, for instance, a non-initialized `complementary-id` and the  `complete_entry` has a valid (an initialized one) we could think on updating it. But this could be addressed in a follow-up.